include/maths/CDataFrameCategoryEncoder.h

```diff
@@ -186,16 +187,37 @@ class MATHS_EXPORT CDataFrameCategoryEncoder final {
     private:
         EEncoding m_Encoding;
         TDoubleVec m_Map;
+
+    public:
+        const TDoubleVec& map() const;
+
+        double fallback() const;
+
+    private:
         double m_Fallback;
         bool m_Binary;
     };
 
+    class Visitor {
+    public:
+        virtual void addOneHotEncoding(std::size_t inputColumnIndex,
+                                       std::size_t hotCategory) = 0;
+        virtual void addTargetMeanEncoding(std::size_t inputColumnIndex,
+                                           const TDoubleVec& map,
+                                           double fallback) = 0;
+        virtual void addFrequencyEncoding(std::size_t inputColumnIndex,
+                                          const TDoubleVec& map) = 0;
+    };
+
 public:
     CDataFrameCategoryEncoder(CMakeDataFrameCategoryEncoder parameters);
 
     //! Initialize from serialized data.
     CDataFrameCategoryEncoder(core::CStateRestoreTraverser& traverser);
 
+    //! Get the complete representation of encodings.
+    const TEncodingUPtrVec& encodings() const;
```

Comment:
Is this still needed now you have visitor?

Reply:
Getter removed.